design/ceph-config-updates.md

```diff
@@ -0,0 +1,215 @@
+  a nonstandard directory by virtue of being sourced via a Kubernetes secret.
+  - The keyring could copied from the secret mount location to Ceph's default location with an init
+    container, and for some daemons, this may be necessary. This should not be done for the mons,
+    however, as the mon keyrings also include the admin keyring, and persisting the admin key to
+    disk should be avoided at all costs for security.
+
+Notable non-required flags:
+- `--{mon,osd,mgr,mds,rgw}-data-dir` settings exist for all daemons, but it is more desirable to use
+  the `/var/lib/ceph/<daemon_type>/ceph-<daemon_id>` directory for daemon data within containers. If
+  possible, mapping the `dataDirHostPath/<rook_daemon_data_dir>` path on hosts to this default
+  location in the container is preferred.
+  - Note that currently, `dataDirHostPath` is mapped directly to containers, meaning that each
+    daemon container has access to other daemon containers' host-persisted data. Modifying Rook's
+    behavior to only mount the individual daemon's data dir into the container as proposed here will
+    be a small security improvement on the existing behavior.
+- `--run-dir` exists for all daemons, but it is likewise more desirable to use the `/var/run/ceph`
+  path in containers. Additionally, this directory stores only unix domain sockets, and it does not
+  need to be persisted to the host. We propose to simply use the `/var/run/ceph` location in
+  containers for runtime storage of the data.
+
+### Additional configuration
+
+Additional configuration which Rook sets up initially should be done by setting values in Ceph's
+centrally-stored config. A large group of additional configurations can be configured at once via
+the Ceph command `ceph config assimilage-conf`. Care should be taken to make sure that Rook does not
+modify preexisting user-specified values.
+
+In the initial version of this implementation, Rook will set these values on every operator restart.
+This may result in user configs being overwritten but will ensure the user is not able to render
+Rook accidentally unusable. In the future, means of determining whether a user has specified a value
+or whether Rook has specified it is desired which may mean a feature addition to Ceph.
+
+Existing global configs configured here:
+- `mon allow pool delete = true`
+- `fatal signal handlers = false`
+  **TODO** is configured here, but this could be a vestigial config from Rook's old, old days
+- `log stderr prefix = "debug "` should be set for all daemons to differentiate logging from auditing
+- `debug ...` configs
+
+Removed configs:
+- `mon_max_pg_per_osd = 1000` is a dangerous setting and should be removed regardless of whether
+  this proposal is accepted
+- `log file = /dev/stderr` is set by default to keep with container standards and kept here if the
+  user needs to change this for debugging/testing
+- `mon cluster log file = /dev/stderr` for `log file` reasons above
```

Comment:
just noticed this.  on the command line, i think we want ``--log-to-stderr --mon-cluster-log-to-stderr --log-stderr-prefix=debug\ ``.  and probably also ``--log-file=`` and ``--mon-cluster-log-file=``.  that way the log file(s) can be turned on (e.g., for debugging) without affecting them always going to stderr for EFK/ELK or whatever logging stack is in place.

Reply:
for non-monitors, it'd just be ``--log-to-stderr --log-stderr-prefix=debug\ ``

Reply:
Hmm... I think what you're suggesting is that the log files be set as blank in the cluster config and the flags `--{,mon-cluster-}log-to-stderr` set true so that by default, no "physical" logs are kept but they are still sent to stdout. In the case where a user sets the log files to a value, then there will be both "physical" and stdout logs. 

I think that sounds like the right idea; "physical" logs being enabled by default would run the risk of filling up disk space in containers rapidly, potentially causing crashes? I have a concern that if logging is enabled for all mons approximately at once, if they log at approximately the same rate and run into a full-disk crash condition, all mons may crash at once. Crash here could also be that they are terminated by Kubernetes. I'm not sure what k8s' behavior here is, but it's something to consider, at least for documentation for users.